samples/bluetooth/mesh/src/main.c

```diff
@@ -107,8 +107,7 @@ static int onoff_status_send(const struct bt_mesh_model *model,
 {
 	uint32_t remaining;
 
-	BT_MESH_MODEL_BUF_DEFINE(buf, OP_ONOFF_STATUS, 3);
-	bt_mesh_model_msg_init(&buf, OP_ONOFF_STATUS);
```

Comment:
Are there still valid use-cases for these two APIs after this change? If not, maybe we should mark them as deprecated?

Reply:
Not really, The `BT_MESH_MODEL_BUF_DEFINE` will also used for other purpose. 

For example, when opcode is not a constant, the Macro defined by this PR does not apply.
Like there: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/large_comp_data_cli.c#L129